bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoAbstractHandler.java

```diff
@@ -282,10 +287,14 @@ protected void disconnected(String message) {
             return miioCom;
         }
         final MiIoBindingConfiguration configuration = getConfigAs(MiIoBindingConfiguration.class);
+        if (configuration.host == null || configuration.host.isEmpty()) {
+            return null;
+        }
         @Nullable
         String deviceId = configuration.deviceId;
         try {
-            if (deviceId != null && deviceId.length() == 8 && tokenCheckPass(configuration.token)) {
+            if (deviceId != null && deviceId.length() == 8 && configuration.token != null
+                    && tokenCheckPass(configuration.token)) {
```

Comment:
Couldn't the `configuration.token != null` be included in the `tokenCheckPass` method?

Reply:
Yes, it could, this was even my first approach and then I changed. But I could revert.